### PR TITLE
Add mspn backbone

### PR DESCRIPTION
**Pull request by wusize**
Add mspn backbone and its unit test.

### CODE REVIEW

File: mmpose/models/backbones/mspn.py [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+            Default: None
+        norm_cfg (dict): dictionary to construct and config norm layer.
+            Default: dict(type='BN')
+    """
+
+    def __init__(self, in_channels, out_channels, **kwargs):
+        super().__init__(in_channels, out_channels * 4, **kwargs)
+
+
+class Downsample_module(nn.Module):
```
Comment (innerlee) [inline mmpose/models/backbones/mspn.py:34]: DownsampleＭodule

File: mmpose/models/backbones/mspn.py [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+            x = module_i(x)
+            if self.has_skip:
+                x = x + skip1[i] + skip2[i]
+            out.append(x)
+        out.reverse()
+
+        return tuple(out)
+
+
+class Upsample_unit(nn.Module):
```
Comment (innerlee) [inline mmpose/models/backbones/mspn.py:114]: UpsampleＵnit

**Comment by wusize** [comment]
Why would this happen?
2020-11-16 15-12-37屏幕截图

**Comment by innerlee** [comment]
network throttling

File: mmpose/models/backbones/mspn.py [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+                                       gen_cross_conv, norm_cfg, in_channels)
+
+    def forward(self, x, skip1, skip2):
+        mid = self.downsample(x, skip1, skip2)
+        out, skip1, skip2, cross_conv = self.upsample(mid)
+
+        return out, skip1, skip2, cross_conv
+
+
+class ResNet_top(nn.Module):
```
Comment (innerlee) [inline mmpose/models/backbones/mspn.py:356]: Naming style

File: mmpose/models/backbones/mspn.py [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+            out.append(outi)
+            skip1.append(skip1_i)
+            skip2.append(skip2_i)
+        skip1.reverse()
+        skip2.reverse()
+
+        return out, skip1, skip2, cross_conv
+
+
+class Single_stage_Network(nn.Module):
```
Comment (innerlee) [inline mmpose/models/backbones/mspn.py:302]: and here

**Review by innerlee — APPROVED** [review]

**Pull request merged**